### DISCUSSION TITLE
What's the best way to incorporate a  Cross-Runtime Library (WinUI) into the new project structure

### DISCUSSION THREAD

**Discussion by ToddGlodek** [unoplatform/uno#12188] [General]
So let's say I have a WinUI & WASM project - and I want to use a cross runtime project to define controls which I will re-use again in other projects.  Let's assume that for a variety of reasons, I need to make a customization to the WASM controls that are in the library.  I setup both the main TestMe & TestMe_Library projects in my solution - and establish a project reference.

<img width="403" alt="TestMe_Setup" src="https://user-images.githubusercontent.com/19764509/236646646-daaa6505-269a-4fb0-9247-ae42e0c7bce2.png">
Then I modify the MyCustomControl in the library and add it to the MainPage in the solution.   Now I want to see the customized control in the browser. 
However, the WASM-specific customization is nor displayed.    

- Are we supposed to create a second TestMe.csproj file that specifically references the TestMe_Library.WASM.csproj?    
- Would that second/alternate project referenced in the main TestMe.Wasm head instead ??

<img width="847" alt="TestMe_Results" src="https://user-images.githubusercontent.com/19764509/236646788-a2f34a60-9738-47be-b2e1-3a0612eebfea.png">

I tried a variety of different approaches - but always ran into a build error related to problems in the auto-generated code. 

What would be super awesome would be to modify the Template Wizard to present the option of including a Cross-runtime library in the newly created solution file.  This would be all pre wired for the various options: Skia/WASM/etc.   This would provide first-time users with an example of the best way to do it.

<img width="992" alt="TestMe_TemplateWizardOption" src="https://user-images.githubusercontent.com/19764509/236647112-f22ad995-b415-4c54-8f7b-9d7b35dad3fc.png">

Perhaps in a future Live-Coding session you could go through the process.   Having a short HOW-TO reference video on the web would also be really useful for first-time Uno Platform users.

**Comment by ToddGlodek** [comment]
I guess it's related to this "HAS_UNO_WASM does not work in latest project template" #11763

**Comment by limeniye** [comment]
About last one.
You can subscribe to the similar issue: https://github.com/unoplatform/uno.templates/issues/503
